alien4cloud-wizard-ui/src/main/webapp/app/features/application-dashboard/deployment-status-piechart/deployment-status-piechart.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, OnInit, ViewEncapsulation, AfterViewInit, ViewChild, Input } from '@angular/core';
+import { Item, DeploymentStatusPiechartDataService } from '@app/core/services/deployment-status-piechart-data.service'
+import { ApplicationEnvironmentDTO } from '@app/core';
+
+@Component({
+  selector: 'w4c-deployment-status-piechart',
+  templateUrl: './deployment-status-piechart.component.html',
+  styleUrls: ['./deployment-status-piechart.component.css']
+})
+export class DeploymentStatusPiechartComponent implements OnInit {
+
+  private map: Map<string, string>;
+  // Data
+  @Input("applicationEnvironmentDTO") applicationEnvironmentDTO: ApplicationEnvironmentDTO[];
+
+  view: any[] = [70, 70];
+
+  // options
+  gradient = false;
+  showLegend = true;
+  showYAxisLabel = true;
+  colorScheme = {
+    domain: []
+  };
+  environments: Array<any> = [];
+  constructor(private deploymentStatusPiechartDataService: DeploymentStatusPiechartDataService) {
+    Object.assign(this, this.environments)   
+  }
+
+  
+
+  ngOnInit() {
+    this.applicationEnvironmentDTO.map(item => {
+      this.environments.push({ "name": item.name+" \n "+item.status, "value": 1})
+      this.getStatusColor(item.status)
+    });
+  }
+
+ 
+  // get status color
+  getStatusColor(status: string){
```

Comment:
???? tu itère sur les clefs d'une map pour en récupérer la valeur !!!!!!! En plus cette map tu la construit dans ton service à chaque appel !!!!!!!!!!!!!! Donc pour résumer pour chaque application, pour chaque environnement, tu instancie une map, et tu Itère dedans pour récupérer ta couleur ? Ya pas un petit pb là ?